opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -235,6 +247,71 @@ public List<DateFormatter> getAllCustomFormatters() {
         .collect(Collectors.toList());
   }
 
+  /**
+   * Retrieves a list of custom formatters and OpenSearch named formatters defined by the user, and
+   * attempts to parse the given date/time string using these formatters.
+   *
+   * @param dateTime The date/time string to parse.
+   * @return A ZonedDateTime representing the parsed date/time in UTC, or null if parsing fails.
+   */
+  public ZonedDateTime getParsedDateTime(String dateTime) {
```

Comment:
we should not choose dateFormatter based on input dateTime.
in getFormattedData, can we move this logic to getFormattedDate?

Reply:
Sorry, can you help me understand more here? In this logic, I'm parsing the input string to date object using OpenSearch field dateFormatters. If we don't want to do this, then we won't be able to parse input string such as `03-Jan-21`(dd-MM-yy) as part of term/range query which causes parsing exception. Aren't we supposed to support all valid date formats that OpenSearch can hold values into as part of Lucene query?

Reply:
Synced offline with @penghuo to keep the parsing date string to date object using OpenSearch dateFormatter logic as it is but to move formatting date string logic to Term/Range query when needed.